### PR TITLE
CI: Added Request review from team

### LINKED ISSUE #444: ✨ Enhancement: Auto-assign 'web-team' to PRs
### Is your feature request related to a problem? Please describe

We have a `json-schema.org/web-team` team that should be assigned to PRs.  This will enable notifications for everyone in the team when a new PR is opened, and anyone in the team can approve/merge the PR.

### Describe the solution you'd like

I remember reading that YAML templates can define assignments, but I can't see to find the docs for it.

A custom action would do fine.  (Ideally we'd want to extend an existing action, if one is available, rather than creating a new one.)

### Describe alternatives you've considered

Manual assignments and tagging are inconsistent.

### Additional context

_No response_

### Are you working on this?

No

### PR DESCRIPTION
**Pull request by aialok**
**What kind of change does this PR introduce?**

- Feature : Added CI for auto requesting review from **TEAM**

**Issue Number:**

-  Closes #444 

**Screenshots/videos:**

**If relevant, did you update the documentation?**

**Summary**

- This workflow will request review from the `web-team` whenever any PR opens : )

**Does this PR introduce a breaking change?**

- No

### CODE REVIEW

**Comment by aialok** [comment]
@gregsdennis @benjagm , GITHUB_TOKEN must have write permission in order to assigned issue.
Because of this workflow get failed.

**Comment by benjagm** [comment]
> @gregsdennis @benjagm , GITHUB_TOKEN must have write permission in order to assigned issue. Because of this workflow get failed.

Can you please AUTH_TOKEN declared as secret in this repo?

**Comment by gregsdennis** [comment]
No, no, GITHUB_TOKEN is a special token and should be all we need.  The action should describe the permissions that it has.

https://docs.github.com/en/actions/security-guides/automatic-token-authentication

**Comment by aialok** [comment]
> No, GITHUB_TOKEN is a special token and should be all we need. The action should describe the permissions that it has.
> 
> https://docs.github.com/en/actions/security-guides/automatic-token-authentication

But for GITHUB_TOKEN , Pull request from public forked repository have only read permission. 

image

Only these token will work. Earlier in my dummy organisation I have used my PAT token with write permission in repo.
Screenshot from 2024-03-06 15-31-20

File: .github/workflows/request-review.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        script: |
+            const Octokit = require('@octokit/core').Octokit;
+            const octokit = new Octokit({
+              request: { fetch: fetch,},
+              auth: process.env.GITHUB_TOKEN,
+            })
+            
+            await octokit.request('POST /repos/{orgs}/{repo}/pulls/{pull_number}/requested_reviewers', {
+              orgs: 'json-schema-org',
+              repo: 'website',
```
Comment (utnim2) [inline .github/workflows/request-review.yml:30]: the `org` name and `repo` name should not be hardcoded use `github.repository_owner` and `github.repository.split('/")[-1]`

[source](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context)

**Review by utnim2 — COMMENTED** [review]
@aialok leave a comment 🙂

**Comment by aialok** [comment]
Well, I have a better way to handle things. We can use the pull_request_target event, which grants both read and write permissions to the GITHUB_TOKEN.

This event runs in the context of the base of the pull request, rather than in the context of the merge commit, as the pull_request event does. This prevents execution of unsafe code from the head of the pull request that could alter your repository or steal any secrets you use in your workflow. This event allows your workflow to do things like label or comment on pull requests from forks. Avoid using this event if you need to build or run code from the pull request.

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target

**Comment by aialok** [comment]
> > @gregsdennis @benjagm , GITHUB_TOKEN must have write permission in order to assigned issue. Because of this workflow get failed.
> 
> Can you please AUTH_TOKEN declared as secret in this repo?

Benjagm, I believe allowing the AUTH_TOKEN to be used by forked repositories, which have both read and write access, as well as contributors outside the organization, could pose a serious security risk. We shouldn't provide such tokens to contributors outside the organization, as they might directly alter our codebase or execute potentially harmful scripts via GitHub Actions. It's imperative that we remove write permissions from the AUTH_TOKEN as soon as possible. Additionally, if any contributor requires write access in a workflow, there are events such as pull_request_target that provide a safer way to handle such permissions.

You can refer to this resource: [Preventing Pwn Requests with GitHub Actions](https://securitylab.github.com/research/github-actions-preventing-pwn-requests/).

**Comment by benjagm** [comment]
> You can refer to this resource

(lets avoid too much tagging). You can create your own personal token for testing purposes, and store it in secret with the name AUTH_TOKEN so you can complete the testing. And when everything is ready just clarify the security scopes required in the token.

File: .github/workflows/request-review.yml (hunk 1 above)
Comment (aialok) [reply .github/workflows/request-review.yml]: Done : )

**Comment by benjagm** [comment]
Hi @aialok . See this [comment](https://github.com/json-schema-org/website/issues/444#issuecomment-1987370037) from  @princerajpoot20. Is it ok to close this PR to implement the other out-of-the-box solution using CODEOWNERS?

**Comment by aialok** [comment]
> Hi @aialok . See this [comment](https://github.com/json-schema-org/website/issues/444#issuecomment-1987370037) from  @princerajpoot20. Is it ok to close this PR to implement the other out-of-the-box solution using CODEOWNERS? 

Yes, we can go with **prince** solution.

**Comment by benjagm** [comment]
Closed as completed with #505 

Thanks a lot @aialok

**Pull request closed without merging**